index.js

```diff
@@ -144,6 +149,31 @@ class HLSSpliceVod {
     });
   }
 
+  insertBumper(bumperMasterManifestUri, _injectBumperMasterManifest, _injectBumperMediaManifest) {
+    return new Promise((resolve, reject) => {
+      this._parseAdMasterManifest(bumperMasterManifestUri, _injectBumperMasterManifest, _injectBumperMediaManifest)
+      .then(bumper => {
+        const bandwidths = Object.keys(this.playlists);
+        for (let b = 0; b < bandwidths.length; b++) {
+          const bw = bandwidths[b];
+
+          const bumperPlaylist = bumper.playlist[findNearestBw(bw, Object.keys(bumper.playlist))];
+          const bumperLength = bumperPlaylist.items.PlaylistItem.length;
+          let i = 0;
+          this.bumperDuration = 0;
+          for (let j = 0; j < bumperLength; j++) {
+            this.playlists[bw].items.PlaylistItem.splice(i + j, 0, bumperPlaylist.items.PlaylistItem[j]);
+            this.bumperDuration += (bumperPlaylist.items.PlaylistItem[j].get('duration') * 1000);
+          }
+          this.playlists[bw].items.PlaylistItem[i + bumperLength].set('discontinuity', true);
+          this.playlists[bw].set('targetDuration', this.targetDuration);
+          this.bumperOffset = bumperLength;
```

Comment:
What is `this.bumperOffset` used for?

Reply:
It is to handle the case where you insert a bumper first and then insert an ad with offset = 0 it should always place it after the video bumper no matter what. So this value stores the offset that the bumper causes all remaining ad inserts.

Reply:
So is it used in insertAdAt( )?
how is this different from `this.bumperDuration`  :O ?
I understand that bumperDuration is in ms and bumperOffset is in number of segments.